addons/point_of_sale/models/res_users.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class ResUsers(models.Model):
+    _inherit = 'res.users'
+
+    @api.multi
+    def unlink(self):
+        for user in self:
+            if (self.env['pos.session'].search_count([('user_id', '=', user.id), ('state', '!=', 'closed')]) 
+                    or user.has_group('point_of_sale.group_pos_user') or user.has_group('point_of_sale.group_pos_user')):
```

Comment:
what do you want to prevent execatly? you say "users that are in open session" but use an `or` clause. If all sessions are closed, I should be able to delete a user with pos group.